index.js

```diff
@@ -1,21 +1,150 @@
 'use strict';
 const {app, BrowserWindow} = require('electron');
+const bufferToArrayBuffer = require('buffer-to-arraybuffer');
 
-module.exports = async (url, {type = 'json', ...options} = {}) => {
+let win;
+let isWinInitialized;
+
+module.exports = async (url, options = {}) => {
 	await app.whenReady();
 
-	const mainWindow = new BrowserWindow({
-		show: false,
-		sandbox: true
-	});
+	if (!win) {
+		win = new BrowserWindow({
+			show: false,
+			sandbox: true
+		});
 
-	await mainWindow.loadURL('about:blank');
+		win.loadURL('about:blank');
+		isWinInitialized = win.webContents.executeJavaScript(
+			ProxyFetch.toString() +
+			'window.proxyFetcher = new ProxyFetch();'
+		);
 
-	const result = await mainWindow.webContents.executeJavaScript(`
-		fetch('${url}', JSON.parse('${JSON.stringify(options)}')).then(x => x.${type}())
-	`, true);
+		isWinInitialized.then(() => {
+			isWinInitialized = true;
+		});
+	}
 
-	mainWindow.destroy();
+	if (isWinInitialized !== true) {
```

Comment:
The `isWinInitialized` logic here is hard to read. I would try to refactor it to be simpler. `isWinInitialized` should not both be a promise and a boolean.

Reply:
Check the recent changes, I just removed the condition.
Initially I added the condition to avoid "overhead" doing an unnecesary `await` but it's probably a bit extreme, so we don't really need the condition.